lib/client.ts

```diff
@@ -122,18 +122,20 @@ class ServiceClientStrictOptions {
  * A custom error returned in case something goes wrong.
  */
 export class ServiceClientError extends Error {
-  constructor (originalError: Error, public type: string, public response?: ServiceClientResponse, name: string = "ServiceClient") {
+  constructor (originalError: Error, public type: string, public response?: ServiceClientResponse, name: string = 'ServiceClient') {
     super(`${name}: ${type}. ${originalError.message || ''}`)
     Object.assign(this, originalError)
   }
 }
 
+const JSON_CONTENT_TYPE_REGEX = /application\/(.*?[+])?json/i
```

Comment:
This RegEx looks a bit funky...

Why is the `+` in a character-class-bracketing? 

And I think, it should be "grounded" (wrapped in `^...$`).

Reply:
I could probably make it escaped, but I kind of visually like character class more. `^` might be ok to add but `$` can't be there https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.17 . It might be possible to use something like `\s*;`

Reply:
ok. then keep it as is